securedrop_client/storage.py

```diff
@@ -344,6 +365,23 @@ def update_missing_files(data_dir: str, session: Session) -> None:
             mark_as_not_downloaded(file.uuid, session)
 
 
+def update_draft_replies(session: Session, source_id: int, timestamp: datetime,
+                         old_file_counter: int, new_file_counter: int) -> None:
+    """
+    When we confirm a sent reply, if there are drafts that were sent after it,
+    we need to reposition them to ensure that they appear _after_ the confirmed
+    replies.
+    """
```

Comment:
I would find it super helpful for you to add a quick description for each parameter. For instance I was confused how the `new_file_counter` was found until I looked at where this function is used. It looks like `new_file_counter` is from the reply (that is causing the draft replies to be reordered) that the server has recorded? And `old_file_counter` is from the reply (that is causing the draft replies to be reordered) that the client has recorded?

Why do we need to set all the draft replies to have the same `file_counter` as the `new_file_counter`? 

It looks like this works but just not sure why you decided to approach it this way.

Reply:
your understanding here is correct - I've added some more explanation and an example in 44c439490770b350ba80144c34b7796ab1d12307, let me know if that makes sense!